cp3pt0-deployment/common/utils.sh

```diff
@@ -502,12 +516,40 @@ function update_cscr() {
         else
             info "Configuring CommonService CR common-service in $namespace"
             ${OC} get commonservice common-service -n "${namespace}" -o yaml | ${YQ} eval '.spec += {"operatorNamespace": "'${operator_ns}'", "servicesNamespace": "'${service_ns}'"}' > common-service.yaml
-            
         fi  
+
         yq eval 'select(.kind == "CommonService") | del(.metadata.resourceVersion) | del(.metadata.uid) | .metadata.namespace = "'${namespace}'"' common-service.yaml | ${OC} apply --overwrite=true -f -
         if [[ $? -ne 0 ]]; then
             error "Failed to apply CommonService CR in ${namespace}"
         fi
+        wait_for_cscr_status "${namespace}" "common-service"
+    done
+
+    rm common-service.yaml
+}
+
+# Check CommonService CR
+function check_cscr() {
+    local operator_ns=$1
+    local service_ns=$2
+    local nss_list=$3
+
+    for namespace in ${nss_list//,/ }
+    do
+        result=$("${OC}" get commonservice common-service -n ${namespace} --ignore-not-found)
+        if [[ -z "${result}" ]]; then
+            info "Creating CommonService CR common-service in $namespace"
+            # copy commonservice from operator namespace
+            ${OC} get commonservice common-service -n "${operator_ns}" -o yaml | ${YQ} eval '.spec += {"operatorNamespace": "'${operator_ns}'", "servicesNamespace": "'${service_ns}'"}' > common-service.yaml
+            
+            yq eval 'select(.kind == "CommonService") | del(.metadata.resourceVersion) | del(.metadata.uid) | .metadata.namespace = "'${namespace}'"' common-service.yaml | ${OC} apply --overwrite=true -f -
+            if [[ $? -ne 0 ]]; then
+                error "Failed to apply CommonService CR in ${namespace}"
+            fi
+        fi
```

Comment:
I believe we do not have to copy master CR(default CS CR in operatorNs) to other namespaces in the scope. This would be done by cs operator if it is working properly.
This function is only used to check CS CR status in the scope. We could check
- The existence of CS CR named `common-service` in each namespace under the scope.
- The status of each CS CR.